tpm2tools/signer.go

```diff
@@ -67,13 +72,23 @@ func (k *Key) GetSigner() (crypto.Signer, error) {
 	if !k.hasAttribute(tpm2.FlagSign) {
 		return nil, fmt.Errorf("non-signing key used with GetSigner()")
 	}
-	if k.pubArea.RSAParameters.Sign == nil {
+
+	var sigScheme *tpm2.SigScheme
+	switch k.pubArea.Type {
+	case tpm2.AlgRSA:
+		sigScheme = k.pubArea.RSAParameters.Sign
+	case tpm2.AlgECC:
+		sigScheme = k.pubArea.ECCParameters.Sign
+	default:
+		return nil, fmt.Errorf("unsupported key type: %v", k.pubArea.Type)
+	}
+	if sigScheme == nil {
```

Comment:
So what I would do here is have the switch statement assign to a "supported algorithm" variable, and then check it below. Like:
```go
var sigScheme *tpm2.SigScheme
var sigAlg tpm2.Algorithm

switch k.pubArea.Type {
case tpm2.AlgRSA:
	sigScheme = k.pubArea.RSAParameters.Sign
	sigAlg = tpm2.AlgRSASSA
case tpm2.AlgECC:
	sigScheme = k.pubArea.ECCParameters.Sign
	sigAlg = tpm2.AlgECDSA
default:
	return nil, fmt.Errorf("unsupported key type: %v", k.pubArea.Type)
}

if sigScheme == nil {
	return nil, fmt.Errorf("key missing required signing scheme")
}
if sigScheme.Alg != sigAlg {
	return nil, fmt.Errorf("unsupported signing algorithm: %v", sigScheme.Alg)
}

hash, err := sigScheme.Hash.Hash()
if err != nil {
	return nil, err
}
return &tpmSigner{k, hash}, nil
```